### PR TITLE
Adds type definitions for raspi-gpio, raspi-i2c, raspi-led, raspi-onewire, raspi-pwm, raspi-serial, raspi-soft-pwm

### PR DESCRIPTION
**Pull request by nebrius**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

**Justification for not checking the first option above:** As discussed in depth at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21618, this suite of packages cannot be installed on anything other than a Raspberry Pi due to install scripts that are Raspberry Pi specific. Without these DefinitelyTyped definitions, the only way to develop using this module on a laptop is to manually copy the definition files into the project and use a custom typings folder, which tends to be kludgy and error-prone IME.

@andy-ms can you add these to the list of excludes once finished, as we discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/22321?

Thanks everyone!

### CODE REVIEW

**Comment by nebrius** [comment]
Thanks @andy-ms!

File: types/raspi-i2c/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Type definitions for raspi-i2c 6.1
+// Project: https://github.com/nebrius/raspi-i2c
+// Definitions by: Bryan Hughes <https://github.com/nebrius>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+/// <reference types="node" />
+import { Peripheral } from 'raspi-peripheral';
+export type IReadCallback = (err: null | Error | string, data: null | Buffer | number) => void;
```
Comment (andy-ms) [inline types/raspi-i2c/index.d.ts]: Nit: Don't begin type names with `I` unnecessarily

Comment (nebrius) [reply types/raspi-i2c/index.d.ts]: Good catch, I missed that one, thanks. Fixed.

**Comment by nebrius** [comment]
Hmm, I'm not sure what's going on with the failed Travis build. Here's the relevant section of the output I see in Travis:

```
Testing...
raspi-soft-pwm OK
raspi-serial OK
raspi-onewire OK
raspi-gpio OK
raspi-i2c OK

travis_time:end:3767b2b1:start=1516138045983308474,finish=1516138127943472669,duration=81960164195
[0K
[32;1mThe command "npm test" exited with 0.[0m

Done. Your build exited with 0.
```

Am I missing part of the output indicating a failure, or is this just a spurious failure?

**Comment by andy-ms** [comment]
Travis was having problems yesterday, just needed to re-run.

**Pull request merged**